Pipeline/CreateUCDComponentVersion/dbb-ucd-packaging.groovy

```diff
@@ -134,31 +145,65 @@ properties.buildReportOrder.each{ buildReportFile ->
 		if(executeRecord.getOutputs().isEmpty() != true) {
 			count += executeRecord.getOutputs().size()
 			executeRecord.getOutputs().each{ output ->
-				def (ds,member) = getDatasetName(output.dataset)
+				def (ds,member) = getContainerAndFile(output.dataset)
 				tempBuildOutputsMap.put(new DeployableArtifact(member, output.deployType) , [ds, buildReport, executeRecord, buildReportRank])
 			}
 		}
 	}
 
+    // store build output information in Hashmap buildOutputsMap to replace potential duplicates
+    // managing TYPE USS_RECORD
+    USSRecords.each{ USSRecord ->
+        ArrayList<ArrayList> outputs = []
+        USSRecord.getAttribute("outputs").split(';').collectEntries { entry ->
+            outputs += entry.replaceAll('\\[|\\]', '').split(',')
+        }
+  
+        if(outputs.size() > 0) {
+            USSCount += outputs.size()
+            outputs.each{ output ->
+                rootDir = output[0].trim()
+                file = output[1].trim()
+                deployType = output[2].trim()
+                tempBuildOutputsMap.put(new DeployableArtifact(file, deployType) , [rootDir, buildReport, USSRecord, buildReportRank])
+            }
+        }
+    }
+
+
 	// store build output information in Hashmap buildOutputsMap to replace potential duplicates
 	// managing DELETE_RECORD leveraging the generic AnyTypeRecord Type
 	deletions.each { deleteRecord ->
 		deletionCount += deleteRecord.getAttributeAsList("deletedBuildOutputs").size()
 		deleteRecord.getAttributeAsList("deletedBuildOutputs").each{ deletedFile ->
-//			def (ds,member) = getDatasetName(deletedFile)
 			String cleansedDeletedFile = ((String) deletedFile).replace('"', '');
 			tempBuildOutputsMap.put(new DeployableArtifact(cleansedDeletedFile, "DELETE") , [deletedFile, buildReport, deleteRecord, buildReportRank])
 		}
 	}
 
-	if ( count + deletionCount == 0 ) {
+	if ( count + USSCount + deletionCount == 0 ) {
```

Comment:
Can we update to use `zosCount`, and `ussCount` as the counter variables?

Reply:
Changed `count `to `datasetMembersCount`, and `USSCount` to `zFSFilesCount` - Hope it's fine like this